src/bin/server.cpp

```diff
@@ -1,36 +1,41 @@
-#include <boost/asio/io_service.hpp>
-
-#include <cstdio>
-#include <cstdlib>
-#include <iostream>
+#include "cxxopts.hpp"
 
 #include "hyrise.hpp"
 #include "scheduler/node_queue_scheduler.hpp"
 #include "server/server.hpp"
-#include "utils/load_table.hpp"
+
+cxxopts::Options get_server_cli_options() {
+  cxxopts::Options cli_options("./hyriseServer", "Starts Hyrise Server in order to accept network requests.");
+
+  // clang-format off
+  cli_options.add_options()
+    ("help", "Display this help and exit")
+    ("p,port", "Specify the port number. 0 means randomly select an available one", cxxopts::value<uint16_t>()->default_value("5432"))  //NOLINT
+    ("execution_info", "Send execution information after statement execution", cxxopts::value<bool>()->default_value("false")) // NOLINT
+    ;  //NOLINT
+  // clang-format on
+
+  return cli_options;
+}
 
 int main(int argc, char* argv[]) {
-  uint16_t port = 5432;
-
-  if (argc >= 2) {
-    char* endptr{nullptr};
-    errno = 0;
-    auto port_long = std::strtol(argv[1], &endptr, 10);
-    Assert(errno == 0 && port_long != 0 && port_long <= 65535 && *endptr == 0, "invalid port number");
-    port = static_cast<uint16_t>(port_long);
+  auto cli_options = get_server_cli_options();
+  const auto parsed_options = cli_options.parse(argc, argv);
+
+  // Print help and exit
+  if (parsed_options.count("help")) {
+    std::cout << cli_options.help() << std::endl;
+    return 0;
   }
 
   // Set scheduler so that the server can execute the tasks on separate threads.
   opossum::Hyrise::get().set_scheduler(std::make_shared<opossum::NodeQueueScheduler>());
 
-  boost::asio::io_service io_service;
-
-  // The server registers itself to the boost io_service. The io_service is the main IO control unit here and it lives
-  // until the server doesn't request any IO any more, i.e. is has terminated. The server requests IO in its
-  // constructor and then runs forever.
-  opossum::Server server{io_service, port};
+  const auto execution_info = parsed_options["execution_info"].as<bool>();
+  const auto port = parsed_options["port"].as<uint16_t>();
 
-  io_service.run();
+  auto server = opossum::Server{port, execution_info};
+  server.run();
```

Comment:
Maybe first setup the server and only if this works, setup the scheduler.

Reply:
Wouldn't that mean that you have to hold all connections until the scheduler is established?

Reply:
Good point. Let's keep it this way.